crux/src/Crux.hs

```diff
@@ -234,20 +234,22 @@ withSelectedOnlineBackend cruxOpts nonceGen selectedSolver maybeExplicitFloatMod
                   , not (yicesMCSat cruxOpts) = ProduceUnsatCores
                   | otherwise                 = NoUnsatFeatures
 
+    extraFeatures = onlineProblemFeatures cruxOpts
+
     withOnlineBackendFM floatRepr =
       case selectedSolver of
-        CCS.Yices -> withYicesOnlineBackend floatRepr nonceGen unsatCoreFeat $ \sym -> do
+        CCS.Yices -> withYicesOnlineBackend floatRepr nonceGen unsatCoreFeat extraFeatures $ \sym -> do
           symCfg sym yicesEnableMCSat (yicesMCSat cruxOpts)
           case goalTimeout cruxOpts of
             Just s -> symCfg sym yicesGoalTimeout (floor s)
             Nothing -> return ()
           k floatRepr sym
-        CCS.CVC4 -> withCVC4OnlineBackend floatRepr nonceGen unsatCoreFeat $ \sym -> do
+        CCS.CVC4 -> withCVC4OnlineBackend floatRepr nonceGen ProduceUnsatCores extraFeatures $ \sym -> do
```

Comment:
It's true that the way `unsatCoreFeat` is written is Yices-specific, but we should still honor the `unsatCores` option, and only ask the solver to produce unsat corses if that option is set.

Reply:
Yikes, good catch! I misread the diff post-merge!